src/main/java/htsjdk/samtools/BAMFileSpan.java

```diff
@@ -115,15 +115,55 @@ public SAMFileSpan removeContentsBefore(final SAMFileSpan fileSpan) {
         validateSorted();
 
         final BAMFileSpan trimmedChunkList = new BAMFileSpan();
+        final long chunkStart = bamFileSpan.chunks.get(0).getChunkStart();
         for(final Chunk chunkToTrim: chunks) {
-            if(chunkToTrim.getChunkEnd() > chunkToTrim.getChunkStart()) {
-                if(chunkToTrim.getChunkStart() >= bamFileSpan.chunks.get(0).getChunkStart()) {
+            if(chunkToTrim.getChunkEnd() > chunkStart) {
```

Comment:
Just want to confirm that you now have at least one unit test that fails before this bug fix (`chunkToTrim.getChunkEnd() > chunkToTrim.getChunkStart()` -> `chunkToTrim.getChunkEnd() > chunkStart`) and passes after it.